testbeam_analysis/tools/plot_utils.py

```diff
@@ -68,6 +68,47 @@ def plot_masked_pixels(input_mask_file, pixel_size=None, dut_name=None, output_p
     if dut_name is None:
         dut_name = os.path.split(input_mask_file)[1]
 
+    # The gui parameter is used to plot from the respective plotting function directly into the graphical
+    # user interface of test_beam analysis. The code is partially hard-coded but works for now.
+    if gui:
+        cmap = cm.get_cmap('viridis')
+        cmap.set_bad('w')
+        c_max = np.percentile(occupancy, 99)
+
+        fig = Figure()
+        _ = FigureCanvas(fig)
+        ax = fig.add_subplot(111)
+        ax.set_title('%s' % (dut_name,))
+        # plot noisy pixels
+        if noisy_pixels is not None:
+            ax.plot(noisy_pixels[:, 1], noisy_pixels[:, 0], 'ro', mfc='none', mec='r', ms=15, mew=1.25, label='Noisy pixels')
+            ax.set_title(ax.get_title() + ',\n%d noisy pixels' % (n_noisy_pixels,))
+        # plot disabled pixels
+        if disabled_pixels is not None:
+            ax.plot(disabled_pixels[:, 1], disabled_pixels[:, 0], 'x', mfc='none', mec='r', ms=15, mew=1.25, label='Disabled pixels')
+            ax.set_title(ax.get_title() + ',\n%d disabled pixels' % (n_disabled_pixels,))
+        ax.imshow(np.ma.getdata(occupancy), aspect=aspect, cmap=cmap, interpolation='none', origin='lower',
+                  clim=(0, c_max))
+        ax.set_xlim(-0.5, occupancy.shape[1] - 0.5)
+        ax.set_ylim(-0.5, occupancy.shape[0] - 0.5)
+        ax.set_xlabel("Column")
+        ax.set_ylabel("Row")
+        leg = ax.legend(numpoints=1, bbox_to_anchor=(1.015, 1.135), loc='upper right')
+        leg.get_frame().set_facecolor('none')
+
+        fig_1 = Figure()
+        _ = FigureCanvas(fig_1)
+        ax = fig_1.add_subplot(111)
+        ax.set_title('%s\n occupancy' % (dut_name,))
+        ax.imshow(occupancy, aspect=aspect, cmap=cmap, interpolation='none', origin='lower', clim=(0, c_max))
+        #     np.ma.filled(occupancy, fill_value=0)
+        ax.set_xlim(-0.5, occupancy.shape[1] - 0.5)
+        ax.set_ylim(-0.5, occupancy.shape[0] - 0.5)
+        ax.set_xlabel("Column")
+        ax.set_ylabel("Row")
+
+        return [fig, fig_1]
+
```

Comment:
The `output_pdf.savefig(fig)` saves the current state of the fig. When changing the figures properties, also the state of the figure is overwritten which leads the both figures `fig` and `fig_1` to be the same. Therefore, when actually returning the figures, they must be 2 different fgures instead of on figure.